src/Model/JoinsTrait.php

```diff
@@ -29,11 +31,19 @@ public function join(string $foreignTable, array $defaults = []): Join
     {
         $defaults[0] = $foreignTable;
 
-        return $this->add(Join::fromSeed($this->_default_seed_join, $defaults));
+        $join = Join::fromSeed($this->_default_seed_join, $defaults);
+
+        if ($this->hasElement($name = $join->getDesiredName())) {
```

Comment:
Can we join same table multiple times?

Reply:
only one Join can be added which can be retrived by `getJoin`, same as for fields and references (even if the setup methods are named `hasOne`/`hasMany`)